src/MessageHandler/AttachEntryEmbedHandler.php

```diff
@@ -67,16 +67,27 @@ public function __invoke(EntryEmbedMessage $message): void
     private function fetchCover(Entry $entry, Embed $embed): ?Image
     {
         if (!$entry->image) {
-            $tempFile = null;
-            if ($embed->image) {
-                $tempFile = $this->fetchImage($embed->image);
-            } elseif ($embed->isImageUrl()) {
-                $tempFile = $this->fetchImage($entry->url);
+            if ($imageUrl = $this->getCoverUrl($entry, $embed)) {
+                if ($tempFile = $this->fetchImage($imageUrl)) {
+                    $image = $this->imageRepository->findOrCreateFromPath($tempFile);
+                    if ($image && !$image->filePath) {
+                        $image->sourceUrl = $imageUrl;
+                    }
```

Comment:
I'd suggest to always write the source url and I am not sure, but I think a `persist` call is needed
```suggestion
                    if ($image) {
                        $image->sourceUrl = $imageUrl;
                        $this->entityManager->persist($image);
                    }
```

Reply:
- it looks like `Entity::$image` is already cascade persisted (https://github.com/MbinOrg/mbin/blob/12ecacfc4d12e597792f0ea2d7316ffcb9425619/src/Entity/Entry.php#L80-L82) so I don't think an extra `persist()` call is needed (and if it is, we should already see its detrimental effects long before the current occasional thumbnail went missing problem)
- always recording source url seems doable, I'll add that, though keep in mind that it might overwrite the previous url if the returned image entity is already known and have a different sourceUrl set, or perhaps worse, unintentionally "downgrading" an uploaded image because the both sides used/reposted the same image, and further down the line the user might not be too pleased to see that their supposedly local uploaded image got swept away from remote image cleaning routine

Reply:
Maybe just setit when there is not one set yet?

Reply:
added, see second commit